app/models/connected_apps/affiliate_sales_data.rb

```diff
@@ -4,7 +4,9 @@
 #
 module ConnectedApps
   class AffiliateSalesData < ConnectedApp
-    def connect; end
+    def connect(_opts)
+      update! data: true # not-nil value indicates it is ready
+    end
```

Comment:
Could we set this value in the constructor so that we don't have to update the record again? Doesn't really matter though.

Reply:
Good thinking